### PR TITLE
fix(client-py): Don't raise exception on 300 status code

### PR DESCRIPTION
**Pull request by ahal**
Issue: #4998

### CODE REVIEW

File: clients/client-py/test/test_client.py [hunk 1]
```diff
@@ -442,6 +442,14 @@ def test_success_fifth_try_connection_errors(client, apiPath):
         assert expected == v
 
 
+def test_redirect_status_code(client, apiPath):
```
Comment (ahal) [inline clients/client-py/test/test_client.py:445]: I was going to add test for the async client, but noticed that there are no test fixtures set up there to make it easy, so decided to skip it. But would be nice to add tests for the async client at some point.

**Review by petemoore — APPROVED** [review]
This looks ok to me, but also happy for others to look over it with more python familiarity and who might actively be using the python client that might be affected (@matt-boris @lotas @djmitche @jschwartzentruber @marco-c).

File: changelog/issue-4998.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+audience: users
+level: patch
```
Comment (petemoore) [inline changelog/issue-4998.md:2]: I'm torn between this being patch and being major.

File: clients/client-py/test/test_client.py (hunk 1 above)
Comment (lotas) [reply clients/client-py/test/test_client.py:445]: 👍 so we don't forget https://github.com/taskcluster/taskcluster/issues/6694

**Review by lotas — APPROVED** [review]

File: changelog/issue-4998.md (hunk 2 above)
Comment (lotas) [reply changelog/issue-4998.md:2]: I'm not sure how much does this change break for existing integrations.

Comment (ahal) [reply changelog/issue-4998.md:2]: The workaround for this bug should keep working after this.

Also in general, it's possible for users to depend on the buggy behaviour of almost any bug, so personally I don't think it should be major.

**Review by matt-boris — APPROVED** [review]
Thanks ahal 👍🏻

**Pull request merged**